src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,71 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.mapper.ProductDtoMapper;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.model.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private ProductService productService;
+    private ProductDtoMapper productDtoMapper;
+
+    public ProductController(ProductService productService, ProductDtoMapper productDtoMapper) {
+        this.productService = productService;
+        this.productDtoMapper = productDtoMapper;
+    }
+
+    @PostMapping
+    public ProductResponseDto createProduct(@RequestBody ProductRequestDto productRequestDto) {
+        Product product = productService.save(productDtoMapper.toModel(productRequestDto));
+        return productDtoMapper.toDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getProduct(@PathVariable Long id) {
+        Product product = productService.get(id);
+        return productDtoMapper.toDto(product);
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteProduct(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto updateProduct(@PathVariable Long id,
+                                            @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productDtoMapper.toModel(productRequestDto);
+        product.setId(id);
+        return productDtoMapper.toDto(productService.save(product));
+    }
+
+    @GetMapping("/byPrice")
+    public List<ProductResponseDto> getByPriceBetween(@RequestParam Double min,
+                                                      @RequestParam Double max) {
+        return productService.getByPriceBetween(min, max).stream()
+                .map(productDtoMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/byCategories")
```

Comment:
```suggestion
    @GetMapping("/by-category")
```
keep in mind proper endpoint naming